firstapp/app/src/androidTest/java/com/example/firstapp/CUJExecutorTest.java

```diff
@@ -0,0 +1,47 @@
+package com.example.firstapp;
+import androidx.test.platform.app.InstrumentationRegistry;
+import androidx.test.uiautomator.By;
+import androidx.test.uiautomator.UiDevice;
+import androidx.test.uiautomator.UiObject2;
+import androidx.test.uiautomator.Until;
+import static org.junit.Assert.*;
+import org.junit.Before;
+import org.junit.Test;
+import java.io.IOException;
+
+public class CUJExecutorTest {
+    private static final String BASIC_SAMPLE_PACKAGE = "com.example.firstapp";
+    private static final int TIMEOUT = 6000;
+
+
+
+    //Basic clicking test
+    @Test
+    public void countToFive() throws Exception {
+        String[] preCUJ = {};
+        String[] postCUJ = {BASIC_SAMPLE_PACKAGE, "click;COUNT", "click;COUNT", "click;COUNT", "click;COUNT", "click;COUNT"};
+        CUJExecutor.executeCUJ(preCUJ, postCUJ);
+        UiObject2 five = ShellUtility.device.wait(Until.findObject(By.textContains("5")),TIMEOUT);
+        assertNotEquals(null, five);
+    }
+    //test Non-empty preCUJ (and shortest acceptable postCUJ), strict matching
+    @Test
+    public void countToFiveMeasureLast() throws Exception {
```

Comment:
Test names should describe what you're testing, instead of putting that description as a comment.  It is okay to have both descriptive test method names and comments.